From 2d2e2f61b61a10d6fb5c828d0354e9d9c1cb03fa Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Tue, 6 Jan 2015 14:33:55 +0100
Subject: [PATCH 3/3] prevent ImageMagick inserting timestamps to PNGs

---
 doc/reference/images/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/doc/reference/images/Makefile.am b/doc/reference/images/Makefile.am
index d1e7f200a5..340af1be03 100644
--- a/doc/reference/images/Makefile.am
+++ b/doc/reference/images/Makefile.am
@@ -1,7 +1,7 @@
 
 PNG = first_level.png construction.png ralD006_1.png
 
-CONVERTOPT = -density 112x112 -units PixelsPerCentimeter
+CONVERTOPT = -density 112x112 -units PixelsPerCentimeter -define png:exclude-chunks=date
 
 # Enable out of source build
 GFXDIR = $(abs_srcdir)/../../../data/gfx48
-- 
2.1.0

