--- enigma-1.01/lib-src/enigma-core/ecl_util.hh~	2006-01-20 12:12:12.000000000 +0100
+++ enigma-1.01/lib-src/enigma-core/ecl_util.hh	2010-02-06 01:53:10.484390523 +0100
@@ -20,6 +20,7 @@
 #define ECL_UTIL_HH_INCLUDED
 
 #include <string>
+#include <algorithm>
 
 /* hide GNU extensions for non-gnu compilers: */
 #ifndef __GNU__
diff -Nur enigma-1.01/build-tree/enigma-1.01/src/Utf8ToXML.cc enigma-1.01.new/build-tree/enigma-1.01/src/Utf8ToXML.cc
--- enigma-1.01/src/Utf8ToXML.cc	2006-09-03 19:15:33.000000000 +0000
+++ enigma-1.01/src/Utf8ToXML.cc	2008-03-18 04:49:15.000000000 +0000
@@ -21,6 +21,7 @@
 #include "main.hh"
 #include <xercesc/util/XMLString.hpp>
 #include <xercesc/util/TransService.hpp>
+#include <cstring>
 
 XERCES_CPP_NAMESPACE_USE
 
diff -Nur enigma-1.01/build-tree/enigma-1.01/src/file.hh enigma-1.01.new/build-tree/enigma-1.01/src/file.hh
--- enigma-1.01/src/file.hh	2007-04-03 23:12:49.000000000 +0000
+++ enigma-1.01/src/file.hh	2008-03-18 04:48:51.000000000 +0000
@@ -22,6 +22,7 @@
 #include <iosfwd>
 #include <vector>
 #include <list>
+#include <memory>
 #include "ecl_error.hh"
 
 namespace enigma
diff -Nur enigma-1.01/build-tree/enigma-1.01/src/gui/widgets.cc enigma-1.01.new/build-tree/enigma-1.01/src/gui/widgets.cc
--- enigma-1.01/src/gui/widgets.cc	2007-05-18 13:14:14.000000000 +0000
+++ enigma-1.01/src/gui/widgets.cc	2008-03-18 04:48:51.000000000 +0000
@@ -28,6 +28,7 @@
 #include <cassert>
 #include <algorithm>
 #include <iostream>
+#include <limits.h>
 
 using namespace enigma::gui;
 using namespace ecl;
--- enigma-1.01/src/lua.cc~	2010-02-06 02:00:59.472140143 +0100
+++ enigma-1.01/src/lua.cc	2010-02-06 02:01:20.695139086 +0100
@@ -572,7 +572,7 @@
     int         y       = round_down<int>(lua_tonumber(L, 2));
     const char *dir     = lua_tostring(L,3);
     const char *allowed = "wsen";
-    char       *found   = strchr(allowed, dir[0]);
+    const char *found   = strchr(allowed, dir[0]);
 
     if (found && found[0]) 
         world::AddScramble(GridPos(x,y), enigma::Direction(found-allowed));
--- enigma-1.01/src/PropertyManager.cc~	2007-05-18 15:14:15.000000000 +0200
+++ enigma-1.01/src/PropertyManager.cc	2010-02-06 02:02:45.056139587 +0100
@@ -29,6 +29,7 @@
 #include "XMLtoUtf8.hh"
 #include "ecl_system.hh"
 #include <iostream>
+#include <cstdio>
 #include <xercesc/dom/DOM.hpp>
 #include <xercesc/util/XMLDouble.hpp>
 #include <xercesc/util/XMLString.hpp>
--- enigma-1.01/src/StateManager.cc~	2010-02-06 02:04:34.963140776 +0100
+++ enigma-1.01/src/StateManager.cc	2010-02-06 02:05:55.729138277 +0100
@@ -33,6 +33,7 @@
 #include "gui/ErrorMenu.hh"
 #include <iostream>
 #include <sstream>
+#include <cstdio>
 #include <xercesc/dom/DOM.hpp>
 #include <xercesc/util/XMLDouble.hpp>
 #include <xercesc/util/XMLString.hpp>
--- enigma-1.01/src/gui/LevelPackComposer.cc~	2007-05-18 15:14:14.000000000 +0200
+++ enigma-1.01/src/gui/LevelPackComposer.cc	2010-02-06 02:07:14.662390121 +0100
@@ -16,7 +16,8 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
  *
  */
- 
+
+#include <cstdio>
 #include "gui/LevelPackComposer.hh"
 #include "gui/HelpMenu.hh"
 #include "ecl.hh"
--- enigma-1.01/src/lev/Index.cc~	2007-05-22 22:02:53.000000000 +0200
+++ enigma-1.01/src/lev/Index.cc	2010-02-06 02:08:15.873139909 +0100
@@ -16,6 +16,7 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
  */
 
+#include <cstdio>
 #include "lev/Index.hh"
 #include "lev/VolatileIndex.hh"
 #include "errors.hh"
